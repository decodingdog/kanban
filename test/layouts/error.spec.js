import { shallowMount } from "@vue/test-utils";
import errorVue from "~/layouts/error.vue";

describe("ErrorLayout", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(errorVue, {
      propsData: {
        error: { statusCode: 404 },
      },
    });
  });

  afterEach(() => {
    if (wrapper) {
      wrapper.destroy();
    }
  });

  test("is Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  test("Mathced Snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("pageNotFound initial value", () => {
    expect(wrapper.vm.pageNotFound).toBe("404 Not Found");
  });

  test("otherError title value", () => {
    expect(wrapper.vm.otherError).toBe("An error occurred");
  });

  test("head title value", () => {
    expect(wrapper.vm.$metaInfo.title).toBe("404 Not Found");
  });
});
