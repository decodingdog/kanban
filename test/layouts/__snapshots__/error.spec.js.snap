// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLayout Mathced Snapshot 1`] = `
"<v-app-stub dark="true" id="app">
  <h1>
    404 Not Found
  </h1>
  <nuxtlink to="/"> Home page </nuxtlink>
</v-app-stub>"
`;
